graylog2-web-interface/src/components/cluster/GraylogClusterOverview.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+/* eslint-disable react/no-find-dom-node */
+
+import * as React from 'react';
+import { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import styled, { css } from 'styled-components';
+import type { DefaultTheme } from 'styled-components';
+import { PluginStore } from 'graylog-web-plugin/plugin';
+
+import EventHandlersThrottler from 'util/EventHandlersThrottler';
+import NumberUtils from 'util/NumberUtils';
+import { useStore } from 'stores/connect';
+import { Col, Row } from 'components/bootstrap';
+import { Spinner } from 'components/common';
+import { ClusterTrafficActions, ClusterTrafficStore } from 'stores/cluster/ClusterTrafficStore';
+import { NodesStore } from 'stores/nodes/NodesStore';
+
+import TrafficGraph from './TrafficGraph';
+
+type Props = {
+  layout: string,
+  children: React.ReactNode
+}
+
+const StyledDl = styled.dl`
+  margin-bottom: 0;
+`;
+const StyledH2 = styled.h2(({ theme }: { theme: DefaultTheme }) => css`
+  margin-bottom: ${theme.spacings.sm};
+`);
+const StyledH3 = styled.h3(({ theme }: { theme: DefaultTheme }) => css`
+  margin-bottom: ${theme.spacings.sm};
+`);
+
+const GraylogClusterOverview = ({ layout, children }: Props) => {
+  const nodes = useStore(NodesStore);
+  const { traffic } = useStore(ClusterTrafficStore);
+  const [graphWidth, setGraphWidth] = useState(600);
+  const eventThrottler = useRef(new EventHandlersThrottler());
+  const containerRef = useRef(null);
+  const licensePlugin = PluginStore.exports('license');
+
+  useEffect(() => {
```

Comment:
This effect mixes up some things and I think we would benefit from splitting it up a bit.